luaui/Widgets/gui_comselect.lua

```diff
@@ -0,0 +1,88 @@
+function widget:GetInfo()
+	return {
+		name = "Commander Selector",
+		desc = "Allows for proper keybind selection of any flavour of commanders",
+		author = "Hornet, with some ZK raiding",
+		date = "03 June 2024",
+		license = "GNU GPL, v2 or later",
+		layer = 1001,
+		enabled = true,
+	}
+end
+
+
+
+local myTeamID = Spring.GetMyTeamID()
+
+local commanderList = {}
+local commanderIds = {}
+for udid, ud in pairs(UnitDefs) do
+	if ud.customParams and ud.customParams.iscommander then
+		commanderIds[udid] = udid
+	end
+end
+
+
+
+function widget:UnitCreated(unitID, unitDefID, unitTeam)
+	if (not myTeamID or unitTeam ~= myTeamID) then
+		return
+	end
+	if commanderIds[unitDefID] then
+		table.insert(commanderList, unitID)
+	end
+end
+
+
+-- comm selection functionality
+local commIndex = 1
+local function SelectComm()
+	local commCount = #commanderList
+
+	if commCount <= 0 then
+		return
+	end
+	
+	-- This check deals with the case of spectators selecting
+	-- teams with different numbers of commanders.
+	if commCount < commIndex then
```

Comment:
specs changing specced team doesn't seem actually handled

Reply:
I took that part from ZK wholesale, in theory it should be as effective here as there, will investigate

Reply:
As far as I can test it works when spectating, no matter the comcount per team, but I am having to test in SP for obvious reasons.

Reply:
And resolved, thanks muchly